storage/replica_test.go

```diff
@@ -1829,6 +1829,52 @@ func TestEndTransactionGC(t *testing.T) {
 	}
 }
 
+// TestEndTransactionResolveOnlyLocalIntents verifies that an end transaction
+// request resolves only local intents.
+func TestEndTransactionResolveOnlyLocalIntents(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	tc := testContext{}
+	tc.Start(t)
+	defer tc.Stop()
+
+	// Split the range and create an intent in each range.
+	// The keys of the two intents are next to each other.
+	key := proto.Key("a")
+	splitKey := key.Next()
+	newRng := splitTestRange(tc.store, splitKey, splitKey, t)
+
+	txn := newTransaction("test", key, 1, proto.SERIALIZABLE, tc.clock)
+	pArgs := putArgs(key, []byte("value"), tc.rng.Desc().RangeID, tc.store.StoreID())
+	pArgs.Txn = txn
+	if _, err := tc.rng.AddCmd(tc.rng.context(), &pArgs); err != nil {
+		t.Fatal(err)
+	}
+
+	pArgs = putArgs(splitKey, []byte("value"), newRng.Desc().RangeID, tc.store.StoreID())
+	pArgs.Txn = txn
+	if _, err := newRng.AddCmd(newRng.context(), &pArgs); err != nil {
+		t.Fatal(err)
+	}
+
+	// End the transaction and resolve the intents.
+	args := endTxnArgs(txn, true /* commit */, 1, tc.store.StoreID())
+	args.Timestamp = txn.Timestamp
+	args.Intents = []proto.Intent{{Key: key, EndKey: splitKey.Next()}}
+	if _, err := tc.rng.AddCmd(tc.rng.context(), &args); err != nil {
+		t.Fatal(err)
+	}
+
+	// Check if the intent in the other range has not yet been resolved.
+	gArgs := getArgs(splitKey, newRng.Desc().RangeID, tc.store.StoreID())
+	_, err := newRng.AddCmd(newRng.context(), &gArgs)
+	if err == nil {
```

Comment:
I think you can skip this. The check for `WriteIntentError` below should get it.